### PR TITLE
[raz] Add JWT support for Hue-Raz Integration

### PR DESCRIPTION
**Pull request by Harshg999**
## What changes were proposed in this pull request?

- Instead of directly asking RAZ for delegation token via Kerberos credentials, we are now fetching JWT from Knox host (this setup can be in HA mode) and then serve that JWT to RAZ to get back the delegation token.
- The later flow using the delegation token remains the same.

## How was this patch tested?

- Updated existing UTs and added new UTs.
- Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
- However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.

### CODE REVIEW

**Comment by Harshg999** [comment]
PR is ready for review.

File: desktop/core/src/desktop/conf.py [hunk 1]
```diff
@@ -2174,6 +2174,28 @@ def has_raz_url():
...
+      help=_('Comma separated host URLs to fetch token from.'),
+      type=str,
+      default='',
+    )
+  )
+)
+
+def is_sdxaas_jwt_enabled():
+  """Check if SDXaaS token url is configured"""
+  return SDXAAS.TOKEN_URL.get() != ''
```
Comment (wing2fly) [inline desktop/core/src/desktop/conf.py:2192]: Nit: Take a look this one: https://stackoverflow.com/questions/9573244/how-to-check-if-the-string-is-empty

**Review by wing2fly — APPROVED** [review]

File: desktop/core/src/desktop/conf.py (hunk 1 above)
Comment (Harshg999) [reply desktop/core/src/desktop/conf.py]: Changed it and using `bool()` now.

**Review by ranade1 — APPROVED** [review]
Looks good to me. Have you tested it end-to-end?

**Review by ranade1 — APPROVED** [review]
Looks good to me. Have you tested it end-to-end?

**Comment by Harshg999** [comment]
> * Updated existing UTs and added new UTs.
> * Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
> * However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.

- Updated existing UTs and added new UTs.
- Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
- However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.

**Comment by Harshg999** [comment]
> Looks good to me. Have you tested it end-to-end?

- Updated existing UTs and added new UTs.
- Tested in a live cluster - Hue is successfully able to fetch JWT from Knox and serve it to RAZ. This was tested for both HA and non-HA cluster.
- However, RAZ is not configured to return delegation token for JWT scenario. So we cannot fully test it E2E yet. This will be tested when RAZ changes are done from their side.

**Pull request merged**